presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java

```diff
@@ -140,11 +140,13 @@ public ConnectorSplitSource getPartitionSplits(ConnectorSession session, Connect
         private final TupleDomain<RaptorColumnHandle> effectivePredicate;
         private final CloseableIterator<ShardNodes> iterator;
 
+        private CompletableFuture<List<ConnectorSplit>> future;
```

Comment:
SourcePartitionedScheduler seems to try to be thread-safe (`schedule` is synchronized), but `close` (which delegates to `splitSource` is not). So, to me, that would seem to require that splitSource be thread-safe (not necessarily *concurrent-safe*)